coverage/src/BamView-WebComponent.js

```diff
@@ -133,32 +138,44 @@ class BamViewChart extends HTMLElement {
         if (this.label) {
             this.shadowRoot.querySelector('#title-text').innerText = this.label;
         }
-
+  
         if (this.broker) {
-            const readDepthPromise = new Promise((resolve, reject) => {
-              this.broker.addEventListener('read-depth', (evt) => {
-                resolve(evt.detail);
-              });
+            document.addEventListener('alignment-data-request', () => {
+                this.toggleSVGContainerAndIndicator(false);
             });
 
-            const headerPromise = new Promise((resolve, reject) => {
-              this.broker.addEventListener('header', (evt) => {
-                resolve(evt.detail);
-              });
+            this.broker.addEventListener('alignment-data', (event) => {
+                const { header, readDepthData } = event.detail;
+                this.bamHeader = header;
+                this.bamReadDepth = readDepthData;
+
+                this.toggleSVGContainerAndIndicator(true);
+                this.updateBamView();
             });
 
-            this.bamReadDepth = await readDepthPromise;
-            this.bamHeader = await headerPromise;
+            document.addEventListener('region-selected', (event) => this.handleGoClick(event.detail));
+            document.addEventListener('gene-entered', (event) => this.handleSearchClick(event.detail));
+
+            this.setupResizeObserver();
+        }
+    }
+
+    toggleSVGContainerAndIndicator(showSVG) {
+        const indicator = this.shadowRoot.querySelector('iobio-loading-indicator');
+        const svgContainer = this.shadowRoot.querySelector('#chart-container');
+        
+        svgContainer.classList.toggle('hidden', !showSVG);
+        indicator.style.display = showSVG ? 'none' : 'block';
+    }
+
+    updateBamView() {
+        // Check if both bamReadDepth and bamHeader are available
+        if (this.bamReadDepth && this.bamHeader) {
```

Comment:
Same concern as above. What's going to happen if either of these aren't available?